### ISSUE TITLE
NO  events.out.tfevents.

### ISSUE THREAD

**Issue by lhj5426** [ultralytics/ultralytics#23231]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Why does YOLO 26 not generate the `events.out.tfevents` file during training?

Without this file, it is impossible to use
`tensorboard --logdir=D:\Ddown\ultralytics-26\runs\detect\train3 --port=6007`
to visualize and monitor the training progress.

In version 11, TensorBoard could be used normally. Why does it not work in version 26?

Is something missing? I have already installed TensorBoard in the environment using `pip install tensorboard`, but the file is still not generated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lhj5426, thank you for your interest in Ultralytics 🚀! This is an automated response to help us triage your issue, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us understand why the `events.out.tfevents` file isn’t being created in your YOLO26 training runs 🧩, please share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that includes:
-Your exact training command (CLI) or a small Python snippet (Python API)
-Your `ultralytics` version and environment details
-A short excerpt of the training console output showing where logs are written (e.g., the `runs/detect/train3` path)
-The directory listing of `runs/detect/train3` after a few epochs (so we can see which files are created)
-If possible, confirm whether the behavior differs across tasks/modes referenced in the [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) docs

Once we have that, we can reproduce on our side and advise next steps ✅

**Comment by raimbekovm** [comment]
This is expected behavior since v8.3.111 (PR #20213) - TensorBoard was disabled by default to speed up training startup.

To enable it:
```bash
yolo settings tensorboard=True
```

After that you should see the `events.out.tfevents` file in your training directory.

Docs: https://docs.ultralytics.com/integrations/tensorboard/

**Comment by lhj5426** [comment]
Got it, that makes sense now. Will enabling TensorBoard slow down the training process significantly?
 
I don’t have an early stopping mechanism set up right now — I save a model checkpoint every 5 epochs and rely on real-time TensorBoard curves to decide when to terminate the training run. It felt a bit off when I didn’t see the TensorBoard logs while running YOLOv26 today.
 
I’ll go configure the settings with the command right away, thanks!

**Comment by glenn-jocher** [comment]
Enabling TensorBoard shouldn’t slow training in any meaningful way in practice since Ultralytics only logs a few scalars once per epoch (the main extra cost is a small one-time graph trace at startup), so it’s fine to keep it on if you rely on live curves; see the [TensorBoard integration docs](https://docs.ultralytics.com/integrations/tensorboard/) for what gets logged. If you want a safety net for stopping without watching curves, you can also enable built-in early stopping via `patience`:

```python
from ultralytics import YOLO

model = YOLO("yolo26n.pt")
model.train(data="coco8.yaml", epochs=300, patience=20, save_period=5, imgsz=640)
```
